test/unit-tests/broker-connection.test.js

```diff
@@ -18,8 +26,52 @@ describe('Broker socket connections', () => {
     });
 
     it('should emit events', (done) => {
-        client.emit('test-conn', (arg) => {
-            expect(arg).to.equal('hello world');
+        client.emit('test:connection', (arg) => {
+            expect(arg).to.equal('dashboard:ping');
+            done();
+        });
+    });
+
+    it('should emit start of new run to dashboard', (done) => {
+        client.emit(
+            'control:new-run',
+            {
+                id,
+                command: 'test:newman-command',
+                startTime: Date.now(),
+            },
+            (type, arg) => {
+                expect(type).to.equal('new-run');
```

Comment:
Not sure what we're testing here? 🤔

Reply:
Right now, just to check that the socket layer is receiving the events emitted to it, it is calling the callback function with the ```type``` of event and arguments it received, which are then being validated here